root/static/scripts/tests/Control/URLCleanup.js

```diff
@@ -2133,6 +2133,17 @@ const {LINK_TYPES, cleanURL, guessType, validationRules} = require('../../edit/U
                      input_entity_type: 'work',
             expected_relationship_type: 'score',
         },
+        // Niconico Video
+        {
+                             input_url: 'https://www.nicovideo.jp/watch/sm2916956',
+                     input_entity_type: 'recording',
+            expected_relationship_type: 'streamingmusic',
+        },
+        {
+                             input_url: 'https://www.nicovideo.jp/user/1050860',
+                     input_entity_type: 'artist',
+            expected_relationship_type: 'social_network',
+        },
```

Comment:
This test is missing `expected_clean_url` (for testing `clean` function output) and `only_valid_entity_types` (for testing `validate` function output).